Tests/Integration/Network/Connection/FogSyncErrorConnectionIntTests.swift

```diff
@@ -0,0 +1,266 @@
+        try testSupportedProtocols(description: description) {
+            try fogConsensusSame(transportProtocol: $0, expectation: $1)
+        }
+    }
+    
+    func fogConsensusSame(
+        transportProtocol: TransportProtocol,
+        expectation expect: XCTestExpectation
+    ) throws {
+        let fogIndex: UInt64 = 61
+        let consensusIndex = fogIndex
+        try attemptRefresh(
+            fogViewBlockIndex: fogIndex,
+            fogLedgerBlockIndex: fogIndex,
+            consensusBlockIndex: consensusIndex,
+            transportProtocol: transportProtocol,
+            expectation: expect,
+            shouldSucceed: true
+        )
+    }
+    
+    func testFogAheadConsensus() throws {
+        try testSupportedProtocols(description: description) {
+            try fogAheadConsensus(transportProtocol: $0, expectation: $1)
+        }
+    }
+    
+    func fogAheadConsensus(
+        transportProtocol: TransportProtocol,
+        expectation expect: XCTestExpectation
+    ) throws {
+        let fogIndex: UInt64 = 321
+        let consensusIndex = fogIndex - 2
+        
+        // Fog ahead of Consensus should succeed (occurs when cached Consensus block info is used)
+        try attemptRefresh(
+            fogViewBlockIndex: fogIndex,
+            fogLedgerBlockIndex: fogIndex,
+            consensusBlockIndex: consensusIndex,
+            transportProtocol: transportProtocol,
+            expectation: expect,
+            shouldSucceed: true
+        )
+    }
+    
+    func testFogBehindConsensusWithinThreshold() throws {
+        try testSupportedProtocols(description: description) {
+            try fogBehindConsensusWithinThresholdAheadConsensus(
+                    transportProtocol: $0,
+                    expectation: $1)
+        }
+    }
+    
+    func fogBehindConsensusWithinThresholdAheadConsensus(
+        transportProtocol: TransportProtocol,
+        expectation expect: XCTestExpectation
+    ) throws {
+        let fogIndex: UInt64 = 5234523462456
+        let consensusIndex = fogIndex + MockFogSyncChecker.threshold - 2
+        
+        // Fog behind but within threshold should succeed
+        try attemptRefresh(
+            fogViewBlockIndex: fogIndex,
+            fogLedgerBlockIndex: fogIndex,
+            consensusBlockIndex: consensusIndex,
+            transportProtocol: transportProtocol,
+            expectation: expect,
+            shouldSucceed: true
+        )
+    }
+    
+    func testFogBehindConsensusAtThreshold() throws {
+        try testSupportedProtocols(description: description) {
+            try fogBehindConsensusAtThreshold(transportProtocol: $0, expectation: $1)
+        }
+    }
+    
+    func fogBehindConsensusAtThreshold(
+        transportProtocol: TransportProtocol,
+        expectation expect: XCTestExpectation
+    ) throws {
+        let fogIndex: UInt64 = 60
+        let consensusIndex = fogIndex + MockFogSyncChecker.threshold - 1
```

Comment:
Doesn't 'AtThreshold' mean that this line should be `let consensusIndex = fogIndex + MockFocSyncChecker.threshold` ?

Reply:
I might need todo some renaming one way or the other here because we're able to get the "consensus **block-count**" (1-N) from the network, but we store it as an **index** (0 - N-1) in the FogSyncChecker.  So in the unit test we -1 to make the block count an index, and all the math assumes we are dealing with indexes, but could just as easily store the consensusBlockCount in the fogSyncChecker and modify the `inSync` calculations with `-1`